fiaas_deploy_daemon/deployer/kubernetes/role_binding.py

```diff
@@ -0,0 +1,100 @@
+                    subject = Subject(kind="ServiceAccount", name=service_account_name, namespace=namespace)
+                    role_binding.roleRef = role_ref
+                    role_binding.subjects = [subject]
+                    self._owner_references.apply(role_binding, app_spec)
+                    role_binding.save()
+                else:
+                    LOG.info(
+                        "Aborting the creation of a roleBinding for Application: %s with %s: %s",
+                        app_spec.name,
+                        role_kind,
+                        role_name
+                    )
+            else:
+                while (True):
+                    role_binding_name = f"{app_spec.name}-{counter}"
+                    try:
+                        role_binding = RoleBinding.get(role_binding_name, namespace)
+                        counter += 1
+                    except NotFound:
+                        role_binding = RoleBinding()
+                        role_binding.metadata = ObjectMeta(
+                            name=role_binding_name, namespace=namespace, labels=custom_labels, annotations=custom_annotations
+                        )
+
+                        role_ref = RoleRef(kind=role_kind, apiGroup="rbac.authorization.k8s.io", name=role_name)
+                        subject = Subject(kind="ServiceAccount", name=service_account_name, namespace=namespace)
+                        role_binding.roleRef = role_ref
+                        role_binding.subjects = [subject]
+                        self._owner_references.apply(role_binding, app_spec)
+                        role_binding.save()
+                        counter += 1
+                        break
+
+    def _find_role_in_role_bindings(self, role_kind, role_name, role_bindings: list[RoleBinding]):
+        for role_binding in role_bindings:
+            if role_binding.roleRef.kind == role_kind and role_binding.roleRef.name == role_name:
+                return role_binding
+            return None
+
+    def _clean_not_needed_role_bindings(self, role_bindings: list[RoleBinding]):
+        for role_binding in role_bindings:
+            if not self._check_if_exists(role_binding):
+                if self._owned_by_fiaas(role_binding):
+                    RoleBinding.delete(role_binding.metadata.name, role_binding.metadata.namespace)
+                    role_bindings.remove(role_binding)
+
+    def _owned_by_fiaas(self, role_binding):
+        return any(
+            ref.apiVersion == "fiaas.schibsted.io/v1" and ref.kind == "Application"
+            for ref in role_binding.metadata.ownerReferences
+        )
+
+    def _check_if_exists(self, role_binding):
+        return (not (role_binding.roleRef.kind == "Role" and role_binding.roleRef.name in self._list_of_roles)
+                and not (role_binding.roleRef.kind == "ClusterRole" and role_binding.roleRef.name in self._list_of_cluster_roles))
```

Comment:
Please add an explanation comment for what this function does, as it's not obvious.
It might make more sense to use a match statement to clarify the cases.

Reply:
I added a comment and changed the method name to `check_if_matches`